### PR TITLE
Configurable port and host

### LINKED ISSUE #39: Allow backend host and Port configurable
The default_host = "su1.3scale.net". 
To work with on-prem version, it is required that you can specify a different host.

### PR DESCRIPTION
**Pull request by avilatusell**
Closes #39

### CODE REVIEW

File: src/client.coffee [hunk 1]
```diff
@@ -20,17 +20,22 @@ parser = new xml2js.Parser
     default_host {String} Optional
   Example:
     Client = require('3scale').Client
-    client = new Client(provider_key, [default_host])
+    client = new Client(provider_key, [default_host], [default_port])
 ###
 
+host = process.env.host || "su1.3scale.net";
+port = process.env.port || 443;
```
Comment (mayorova) [inline src/client.coffee]: I think it's a bit dangerous to use these env vars, as they may be used for other purposes (for the PORT and HOST of the API app itself, for example). Maybe we need something more specific? (`threescale_host`, `threescale_port` or similar)

Comment (mikz) [inline src/client.coffee]: It should not use ENV vars. Just pass it from the constructor.

File: test/client_test.coffee [hunk 2]
```diff
@@ -15,14 +15,22 @@ describe 'Basic test for the 3Scale::Client', ->
...
       client = new Client(123)
       assert.equal client.host, 'su1.3scale.net'
 
     it 'can change the default host', ->
       client = new Client(123, 'example.com')
       assert.equal client.host, 'example.com'
 
+    it 'should have a default port', ->
+      client = new Client(123, 'example.com')
+      assert.equal client.port, '443'
```
Comment (mikz) [inline test/client_test.coffee]: It is not an issue that the port is a string? I guess the underlying library converts it to number, right?

File: src/client.coffee [hunk 3]
```diff
@@ -20,17 +20,23 @@ parser = new xml2js.Parser
...
-    client = new Client(provider_key, [default_host])
+    client = new Client(provider_key, [default_host], [default_port])
 ###
 
+
 module.exports = class Client
   DEFAULT_HEADERS: { "X-3scale-User-Agent": "plugin-node-v#{VERSION}" }
 
-  constructor: (provider_key, default_host = "su1.3scale.net") ->
+  constructor: (provider_key, provider_host, provider_port) ->
```
Comment (davidor) [inline src/client.coffee]: Nitpick: I think that `provider_host` and `provider_port` are misleading name vars. They are referring to 3scale's service management API host and port, not to anything related to the provider. I think that in this case using `host` and `port` is fine.

File: test/client_test.coffee [hunk 4]
```diff
@@ -15,14 +15,22 @@ describe 'Basic test for the 3Scale::Client', ->
...
     it 'can change the default host', ->
       client = new Client(123, 'example.com')
       assert.equal client.host, 'example.com'
 
+    it 'should have a default port', ->
+      client = new Client(123, 'example.com')
+      assert.equal client.port, 443
+
+    it 'can change the default port', ->
+      client = new Client(123, 'example.com', '3000')
```
Comment (davidor) [inline test/client_test.coffee]: I think '3000' should be an integer for clarity.

Comment (avilatusell) [inline test/client_test.coffee]: OK

**Pull request merged**